docs/server-core/Templates/_CustomAdapters.mdx

```diff
@@ -0,0 +1,236 @@
+## Custom Adapters
+
+The PHP Core SDK provides two base adapter classes that allow you to customize how configuration specs are fetched and how events are logged. This is particularly useful for integrating with existing infrastructure or implementing custom caching strategies.
+
+### SpecsAdapterBase - Custom Configuration Sources
+
+The `SpecsAdapterBase` allows you to fetch configuration specs from custom sources instead of the default Statsig servers. This is useful when you want to:
+
+- Cache specs in a database or Redis
+- Fetch specs from a custom API endpoint
+- Implement custom retry logic or fallback mechanisms
+- Use existing configuration management systems
+
+**Note:** You can get a copy of your current specs data by visiting: `https://api.statsigcdn.com/v2/download_config_specs/{YOUR_SDK_KEY}.json`
+
+#### Implementation
+
+To create a custom specs adapter, extend the `SpecsAdapterBase` class and implement the required methods. The following example shows how to create a database-backed specs adapter:
+
+```php
+<?php
+
+use Statsig\SpecsAdapterBase;
+use Statsig\SpecsUpdateListener;
+
+class DatabaseSpecsAdapter extends SpecsAdapterBase
```

Comment:
Instead of a Database example, add a Redis example that just returns the latest JSON string found for "statsig_config_specs"